Sources/SwiftProtobuf/BinaryEncoder.swift

```diff
@@ -19,30 +19,35 @@ import Foundation
  * Encoder for Binary Protocol Buffer format
  */
 internal struct BinaryEncoder {
-    private var pointer: UnsafeMutablePointer<UInt8>
+    private var pointer: UnsafeMutableRawPointer
 
-    init(forWritingInto pointer: UnsafeMutablePointer<UInt8>) {
+    init(forWritingInto pointer: UnsafeMutableRawPointer) {
         self.pointer = pointer
     }
 
     private mutating func append(_ byte: UInt8) {
-        pointer.pointee = byte
-        pointer = pointer.successor()
+        pointer.storeBytes(of: byte, as: UInt8.self)
+        pointer = pointer.advanced(by: 1)
     }
 
     private mutating func append(contentsOf data: Data) {
-        let count = data.count
-        data.copyBytes(to: pointer, count: count)
-        pointer = pointer.advanced(by: count)
+        data.withUnsafeBytes { dataPointer in
+            if let baseAddress = dataPointer.baseAddress, dataPointer.count > 0 {
```

Comment:
the next helper used a local for count to avoid fetching it a few times, any reason to do it there and not here?

Reply:
Yeah, `UnsafeRawBufferPointer.count` has [been `inlinable` since Swift 4.0 at least](https://github.com/apple/swift/blob/3416108abd0df3997a533a14c5ad06de20ba8a60/stdlib/public/core/UnsafeRawBufferPointer.swift.gyb#L498-L504), so the compiler should observe that the load is trivial. This was not necessarily true of `Data.count`, and so storing the local to avoid the re-load was potentially worth it.

I can restore the local if you would like, but I wouldn't expect a meaningful performance difference.

Reply:
It was more the consistence of the code that was what I noticed.

Reply:
I’m happy to be guided by your preferences here.